tools/autograd/derivatives.yaml

```diff
@@ -2959,3 +2959,6 @@
 - name: _reshape_copy(Tensor self, SymInt[] size) -> Tensor
   self: grad.reshape_symint(self.sym_sizes())
   result: auto_linear
+
+- name: all_reduce(Tensor self, str reduceOp, str tag, int[] ranks, int stride) -> Tensor
+  self: grad
```

Comment:
i think this is wrong, i think it should be ones_like(grad)?  need to do the math and confirm..

Reply:
Hmm It should be an all_gather I think.

Reply:
It's a bit more complicated as each reduction requires a different grad formula.

I'll work on it in parallel with order issues.

Reply:
Removed backwards for now given only "sum" is easy to support.